udata_front/theme/gouvfr/assets/js/pages/Admin/Members/Members.vue

```diff
@@ -0,0 +1,274 @@
+import AdminDangerZone from "../../../components/AdminDangerZone/AdminDangerZone.vue";
+import Avatar from "../../../components/discussions/Avatar/Avatar.vue";
+import Breadcrumb from "../../../components/Breadcrumb/Breadcrumb.vue";
+import SelectGroup, { type Option } from "../../../components/Form/SelectGroup/SelectGroup.vue";
+import MembershipRequest from "../../../components/MembershipRequest/MembershipRequest.vue";
+import type { EditingMember, Member, MemberRole, PendingMembershipRequest } from "../../../types";
+import { useToast } from "../../../composables/useToast";
+
+const props = defineProps<{oid: string}>();
+
+const { t } = useI18n();
+
+const { toast } = useToast();
+
+const membershipRequests = ref<Array<PendingMembershipRequest>>([]);
+
+const roles = ref<Array<Option>>([]);
+
+// TODO : use `useCurrentOrganization` when merged
+const currentOrganization = {name: "SomeName"};
+
+const members = ref<Array<EditingMember>>([]);
+
+const loading = ref(false);
+
+function getModalId(id: string) {
+  return "fr-modal-user-" + id;
+}
+
+function getModalTitleId(id: string) {
+  return "fr-modal-title-user-" + id;
+}
+
+function closeModal(memberId: string) {
+  const modal = document.getElementById(getModalId(memberId));
+  globalThis.dsfr(modal).modal.conceal();
+}
+
+async function updateRole(member: EditingMember) {
+  if(!member.newRole) {
+    closeModal(member.user.id);
+    return;
+  }
+  try {
+    loading.value = true;
+    await updateMemberRole(props.oid, member.user.id, member.newRole);
+    updateMembers();
+    closeModal(member.user.id);
+  } catch (e) {
+    toast.error(t("An error occurred while update member role."));
+  } finally {
+    loading.value = false;
+  }
+}
+
+async function removeMemberFromOrganization(member: Member) {
+  try {
+    loading.value = true;
+    await removeMember(props.oid, member.user.id);
+    updateMembers();
+  } catch (e) {
+    toast.error(t("An error occurred while removing this member."));
+  } finally {
+    loading.value = false;
+  }
+}
+
+function getRoleLabel(role: MemberRole) {
+  return roles.value.find(memberRole => memberRole.value === role)?.label ?? role;
+}
+
+function getRolesWithSelected(role: MemberRole) {
+  return roles.value.map(memberRole => {
+    const updatedMemberRole = {...memberRole};
+    if(updatedMemberRole.value === role) {
+      updatedMemberRole.selected = true;
+    }
+    return updatedMemberRole;
+  });
+}
+
+async function updateMembers() {
+  const organization = await getOrganization(props.oid);
+  members.value = organization.members;
+}
+
+async function updateMemberships() {
+  const memberships = await getPendingMemberships(props.oid);
+  membershipRequests.value = memberships;
+}
+
+async function accept(id: string) {
+  loading.value = true;
+  const promises = [];
+  try {
+    await acceptRequest(props.oid, id);
```

Comment:
I think we may be missing a `catch` and toast for accept and refuse membership request